### PR TITLE
Added session_mode=token query parameter

### LINKED ISSUE #18: Android client should add token query string parameter to login request
?session_mode=token

### PR DESCRIPTION
**Pull request by mamaso**
resolves #18 

@mattchenderson

### CODE REVIEW

File: sdk/test/sdk.testapp/src/main/java/com/microsoft/windowsazure/mobileservices/sdk/testapp/test/LoginTests.java [hunk 1]
```diff
@@ -184,9 +182,7 @@ private void testLoginOperationWithParameter(final Object provider) throws Throw
         }
 
         // Assert
-        String urlPrefix = ".auth/login/";
-        String expectedURL = appUrl + urlPrefix + provider.toString().toLowerCase(Locale.getDefault()) + parameterQueryString;
-        assertEquals(expectedURL, result.getRequestUrl());
+        assertEquals(buildExpectedUrl(provider, parameters), result.getRequestUrl());
```
Comment (phvannor) [inline sdk/test/sdk.testapp/src/main/java/com/microsoft/windowsazure/mobileservices/sdk/testapp/test/LoginTests.java]: I would like a test that verifies behavior if the user adds session_mode = cookie manually, what ends up in the final product.  For now, we said token wins.  (And we aren't worried on casing, but if you want to strip other cases that is ok as well)

File: sdk/src/sdk/src/main/java/com/microsoft/windowsazure/mobileservices/authentication/LoginManager.java [hunk 2]
```diff
@@ -574,6 +574,16 @@ protected void onPostExecute(ServiceFilterResponse response) {
         return future;
     }
 
+    private String normalizeParameters(HashMap<String, String> parameters) {
+        if (parameters == null) {
+            parameters = new HashMap<String, String>();
+        } else {
+            parameters = new HashMap<String, String>(parameters);
+        }
+        parameters.put("session_mode", "token");
```
Comment (phvannor) [inline sdk/src/sdk/src/main/java/com/microsoft/windowsazure/mobileservices/authentication/LoginManager.java]: This should not be sent on client directed flows ideally, they don't have such a concept

File: sdk/test/sdk.testapp/src/main/java/com/microsoft/windowsazure/mobileservices/sdk/testapp/test/LoginTests.java (hunk 1 above)
Comment (mamaso) [inline sdk/test/sdk.testapp/src/main/java/com/microsoft/windowsazure/mobileservices/sdk/testapp/test/LoginTests.java]: good call, I'll add one

File: sdk/test/sdk.testapp/src/main/java/com/microsoft/windowsazure/mobileservices/sdk/testapp/test/LoginTests.java [hunk 3]
```diff
@@ -426,4 +453,23 @@ public void testLogout() {
...
+        String paramString = "";
+        for (Map.Entry<String, String> parameter : params.entrySet()) {
+            if (paramString == "") {
+                paramString = "?";
+            } else {
+                paramString += "&";
+            }
+            paramString += parameter.getKey() + "=" + parameter.getValue();
+        }
+
```
Comment (soninaren) [inline sdk/test/sdk.testapp/src/main/java/com/microsoft/windowsazure/mobileservices/sdk/testapp/test/LoginTests.java]: Unit test code is same as code in SDK , replace the HashMap<String, String> params with string params to verify the created string.

Comment (mamaso) [inline sdk/test/sdk.testapp/src/main/java/com/microsoft/windowsazure/mobileservices/sdk/testapp/test/LoginTests.java]: I verified the expected strings manually during testing so I'd prefer to use the generic method for building urls, is that fine with you?

**Pull request merged**